packages/tests/server/react/getQueryKey.test.tsx

```diff
@@ -39,6 +53,171 @@ const ctx = konn()
   .done();
 
 describe('getQueryKeys', () => {
+  test('no input', async () => {
+    const { proxy, App } = ctx;
+
+    function MyComponent() {
+      const happy1 = getQueryKey(proxy.post.all, undefined, 'query');
+      const happy2 = getQueryKey(proxy.post.all);
+
+      // @ts-expect-error - post.all has no input
+      const sad = getQueryKey(proxy.post.all, 'foo');
+
+      return (
+        <>
+          <pre data-testid="qKey1">{JSON.stringify(happy1)}</pre>
+          <pre data-testid="qKey2">{JSON.stringify(happy2)}</pre>
+        </>
+      );
+    }
+
+    const utils = render(
+      <App>
+        <MyComponent />
+      </App>,
+    );
+
+    await waitFor(() => {
+      expect(utils.getByTestId('qKey1')).toHaveTextContent(
+        JSON.stringify([['post', 'all'], { type: 'query' }]),
+      );
+      expect(utils.getByTestId('qKey2')).toHaveTextContent(
+        JSON.stringify([['post', 'all']]),
+      );
+    });
+  });
+
+  test('with input', async () => {
+    const { proxy, App } = ctx;
+
+    function MyComponent() {
+      const happy1 = getQueryKey(proxy.post.byId, { id: 1 }, 'query');
+
+      // doesn't really make sense but should still work
+      const happyIsh = getQueryKey(proxy.post.byId, { id: 1 });
+
+      // @ts-expect-error - post.byId has required input
+      const sad = getQueryKey(proxy.post.byId, undefined, 'query');
+
+      return (
+        <>
+          <pre data-testid="qKey1">{JSON.stringify(happy1)}</pre>
+          <pre data-testid="qKey2">{JSON.stringify(happyIsh)}</pre>
+        </>
+      );
+    }
+
+    const utils = render(
+      <App>
+        <MyComponent />
+      </App>,
+    );
+
+    await waitFor(() => {
+      expect(utils.getByTestId('qKey1')).toHaveTextContent(
+        JSON.stringify([['post', 'byId'], { input: { id: 1 }, type: 'query' }]),
+      );
+      expect(utils.getByTestId('qKey2')).toHaveTextContent(
+        JSON.stringify([['post', 'byId'], { input: { id: 1 } }]),
+      );
+    });
+  });
+
+  test('mutation', async () => {
+    const { proxy, App } = ctx;
+
+    function MyComponent() {
+      const happy = getQueryKey(proxy.post.update, { name: 'trash' });
+
+      return (
+        <div>
+          <pre data-testid="qKey">{JSON.stringify(happy)}</pre>
+        </div>
+      );
+    }
+
+    const utils = render(
+      <App>
+        <MyComponent />
+      </App>,
+    );
+
+    await waitFor(() => {
+      expect(utils.getByTestId('qKey')).toHaveTextContent(
+        JSON.stringify([['post', 'update'], { input: { name: 'trash' } }]),
+      );
+    });
+  });
```

Comment:
This is wrong. Mutation keys don't take input (since that's now known when calling the hook - just when you do `mutate`)